x/mint/keeper/msg_server.go

```diff
@@ -25,13 +25,21 @@ func NewMsgServerImpl(k Keeper) types.MsgServer {
 }
 
 // UpdateParams updates the params.
-func (ms msgServer) UpdateParams(goCtx context.Context, req *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
-	if ms.authority != req.Authority {
-		return nil, errors.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", ms.authority, req.Authority)
+func (ms msgServer) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams) (*types.MsgUpdateParamsResponse, error) {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
```

Comment:
If we check the authority equality after, do we really need to verify if the input is a valid address? I actually do not think so, imho the inequality error message will be sufficiently clear.

Reply:
isnt the one stored on disk decoded as well?

Reply:
oh it seems in this module we dont. there seems to be a mix of some modules decoding and others not. Which should we go with?

Reply:
Not sure I follow. We always check the given authority compared to the module authority.

The address validity check is useful is CLI, but my point was if you submit a transaction with an address "foo", that it returns an error with "foo is an invalid address" or "expected cosmos1abc, got: foo" are both explicit, so we could remove that first check in the msg server.

Reply:
I have the same opinion as julien, as we always have checks for module authority. It covers the address validity check too returning "invalid authority".

Reply:
Alright, let me submit a PR simplifying that for all modules afterward.